src/cache_cf.cc

```diff
@@ -515,7 +515,8 @@ parseOneConfigFile(const char *file_name, unsigned int depth)
         const char* append = tmp_line_len ? skip_ws(config_input_line) : config_input_line;
 
         size_t append_len = strlen(append);
-
+        if(tmp_line_len+append_len >= std::numeric_limits<int>::max())
+            fatalf("Long line in config file");
         tmp_line = (char*)xrealloc(tmp_line, tmp_line_len + append_len + 1);
```

Comment:
Please fix the `tmp_line_len` type to become `size_t` and then use the recently added (commit 7224761) Sum() to properly handle overflows. Here is an untested sketch for the second part:

```suggestion
        if (const auto combinedSize = Sum(tmp_line_len, append_len, 1U))
            tmp_line = static_cast<char*>(xrealloc(tmp_line, combinedSize.value()));
        else
            throw TextException("Exceedingly long line continuation", Here());
```